modules/dcache/src/main/java/org/dcache/services/ssh2/Ssh2Admin.java

```diff
@@ -318,6 +331,71 @@ public boolean authenticate(String userName, PublicKey key,
         }
     }
 
+    private static class RemoteHostValidator {
+
+        private enum Outcome {
+            ALLOW, DENY, DEFER
+        }
+
+        private boolean isValidHost(String line, ServerSession session) {
+            for (String linePart : Splitter.on(" ").trimResults().omitEmptyStrings().split(line)) {
+                if (linePart.startsWith("from=") && linePart.endsWith("\"")) {
+                    Set<Outcome> outcomes = EnumSet.noneOf(Outcome.class);
```

Comment:
The code is fine as-is, but you could also get by with just a boolean to record whether Outcome.ALLOW has been returned, provided a Outcome.DENY short-circuits the tests and immediate returns false.